### ISSUE TITLE
Log2Lin's lin2log math might be incorrect

### ISSUE THREAD

**Issue by markreidvfx** [NatronGitHub/openfx-misc#76]
Hi, 

I was looking at the implementation of log2lin for reference and I think the lin2log math might be incorrect.

https://github.com/NatronGitHub/openfx-misc/blob/master/Log2Lin/Log2Lin.cpp#L146

the comment there says

```
offset = pow(10,(blackpoint - whitepoint) * 0.002 / gamma)
gain = 1/(1-offset)
linear = gain * (pow(10,(1023*v - whitepoint)*0.002/gamma) - offset)
cineon = (log10((v + offset) /gain)/ (0.002 / gamma) + whitepoint)/1023
```
To me, the should cineon lin2log expression be

```
cineon = (log10(v/gain + offset )/ (0.002 / gamma) + whitepoint)/1023
```

`v/gain + offset` instead of  `(v + offset)/gain`

Also the comment shows up twice in the file and the first one is different

**Comment by devernay** [comment]
There is definitely an inconsistency between the two formulas.
Note that TuttleOFX has the same issue:
- https://github.com/tuttleofx/TuttleOFX/blob/36fc4cae15092a84ea8c29b9c6658c7cabfadb6e/libraries/terry/src/terry/colorspace/gradation.hpp#L348
- https://github.com/tuttleofx/TuttleOFX/blob/36fc4cae15092a84ea8c29b9c6658c7cabfadb6e/libraries/terry/src/terry/colorspace/gradation.hpp#L381

I can't find where I got these formulas from. Do you have a reference? Maybe the log2lin formula is wrong, and not the lin2log?

**Comment by devernay** [comment]
This doc, which is to be the foundation to Cineon, seems to imply that log2lin is wrong, rather than lin2log:
http://www.dotcsw.com/doc/cineon1.pdf
<img width="639" alt="image" src="https://user-images.githubusercontent.com/3778071/92534699-52ae0900-f1ea-11ea-8b98-eda83cc2a03b.png">
<img width="640" alt="image" src="https://user-images.githubusercontent.com/3778071/92534716-60638e80-f1ea-11ea-8a39-b987854cc880.png">
<img width="792" alt="image" src="https://user-images.githubusercontent.com/3778071/92534776-88eb8880-f1ea-11ea-90de-721f3069f666.png">

**Comment by devernay** [comment]
Does this look good to you? https://github.com/NatronGitHub/openfx-misc/commit/c89ac63b975de2e56978f7e59b69455ec0bc775f

**Comment by markreidvfx** [comment]
The inverse math and the spec make sense to me!

I thought the lin2log was wrong because the log2lin matches nuke.

But the spec implies nuke's math is wrong

```
(pow(10,(1023*x-685)/300)-.0108)/(1-.0108)
```
they shuffled a couple things around, divides instead of multiplies

```
1 /0.002 / 0.6 == 300
offset == 0.0108
``` 
The minus offset is happening before the gain.

am I crazy?

**Comment by devernay** [comment]
@markreidvfx I see that [python colour' version](https://github.com/colour-science/colour/blob/162631dcdcb9e289cbf2d821d005f6bca52aa328/colour/models/rgb/transfer_functions/cineon.py) referes to [This formula in Sony's code](https://github.com/imageworks/OpenColorIO-Configs/blob/master//nuke-default/make.py#L163)

**Comment by markreidvfx** [comment]
Yes I agree, I think it might be best fix the lin2log like I initially suggested. [python colour's lin2log](https://github.com/colour-science/colour/blob/162631dcdcb9e289cbf2d821d005f6bca52aa328/colour/models/rgb/transfer_functions/cineon.py#L79) version seems to also match the one I formulated by inverting the log2lin.  

I still find it a bit un-nerving that the spec says otherwise, but I guess it is better to be compatible other implementations.

**Comment by devernay** [comment]
Does https://github.com/NatronGitHub/openfx-misc/pull/79 look right?

**Comment by devernay** [comment]
closed by #79

**Comment by markreidvfx** [comment]
sorry, I missed this, looks good to me!
